br/pkg/lightning/restore/restore.go

```diff
@@ -1236,7 +1255,54 @@ func (rc *Controller) buildRunPeriodicActionAndCancelFunc(ctx context.Context, s
 
 var checksumManagerKey struct{}
 
+const (
+	pauseGCTTLForDupeRes       = time.Hour
+	pauseGCIntervalForDupeRes  = time.Minute
+	pauseGCServiceIDForDupeRes = "lightning-duplicate-resolution"
+)
+
+func (rc *Controller) pauseGCOnce(ctx context.Context, pdCli pd.Client) error {
+	// Get current GC safe point.
+	safePoint, err := pdCli.UpdateGCSafePoint(ctx, 0)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	// Pause GC for 1 hour.
+	_, err = pdCli.UpdateServiceGCSafePoint(ctx,
+		pauseGCServiceIDForDupeRes, int64(pauseGCTTLForDupeRes/time.Second), safePoint)
```

Comment:
* Please make sure there will be at most only one lightning instance that executes this procedure, otherwise you might need to attach an UUID to the service id to avoid them colliding with each other.
* Please check the return value. The return value is the minimum service safepoint currently in this cluster, which means, if it's value is greater than your specified value, it means you failed to register this service safepoint.
* It's not a good idea to use the value fetched from `UpdateGCSafePoint` as the service safepoint. They are two different things. And it cannot guarantee later GC will definitely be prevented.
  * If you only need to protect your new data after some timestamp `T`, this code works, but please consider using `T - 1`.
  * If you do need to let TiDB stop GC no matter what safepoint it is, please consider using the minimum service safepoint as the value, which can be find by invoking `pdCli.UpdateServiceGCSafePoint(ctx, pauseGCServiceIDForDupeRes, int64(pauseGCTTLForDupeRes/time.Second), 1)`. Also note that you have no way to stop an already-started GC.

Reply:
Thanks for your review! I have another question. Can I assume the safepoint is registered successfully if the minimum service safepoint is less or **equal** than the specified value.

Reply:
It seems that every lightning instance will  call `keepPauseGCForDupeRes` before restore tables.